php/Block.php

```diff
@@ -63,66 +63,179 @@ public function register_block() {
 	 * @return string The markup of the block.
 	 */
 	public function render_callback( $attributes, $content, $block ) {
-		$post_types = get_post_types(  [ 'public' => true ] );
-		$class_name = $attributes['className'];
-		ob_start();
 
+		// Store this in a local var for caching since we'll reference it a few
+		// times in this method.
+		$current_post_id = get_the_ID();
+
+		/**
+		 * Fetch all public post types as objects, and construct a new array that
+		 * maps the post type's label name (not slug), and the number of
+		 * published posts within that post type so we can easily display it
+		 * below.
+		 */
+
+		// Array to store our new values in.
+		$public_post_type_counts = [];
+
+		// Get all public post type objects.
+		$post_type_objects = get_post_types(  [ 'public' => true ], 'objects' );
+
+		// Validate the values.
+		if (
+			! empty( $post_type_objects )
+			&& is_array( $post_type_objects )
+		) {
+
+			// Loop through each result and store the label and count mapped to
+			// the slug.
+			foreach ( $post_type_objects as $post_type_object ) {
+				/**
+				 * @var array $public_post_type_counts = [
+				 *     'posts' => [
+				 *         'label' => 'Posts',
+				 *         'count' => 5,
+				 *      ],
+				 *      'pages' => [
+				 *        'label' => 'Pages',
+				 *        'count' => 3,
+				 *      ],
+				 * ]
+				 */
+				$public_post_type_counts[ $post_type_object->name ] = [
+					'label' => $post_type_object->labels->name ?? '',
+					'count' => wp_count_posts( $post_type_object->name )->publish ?? 0,
+				];
+			}
+		}
+
+		/**
+		 * Query for up to 5 posts & pages published between 9:00 and 17:00,
+		 * with the tag (slug) `foo`, and category (name) `baz`.
+		 */
+		$posts_list_query = new WP_Query(
+			[
+				'post_type'              => [ 'post', 'page' ],
+				'post_status'            => 'publish',
+				'fields'                 => 'ids',
+				'update_post_meta_cache' => false, // We're not displaying any post meta.
+				'update_post_term_cache' => false, // Or terms.
+				'posts_per_page'         => 6, // Actual number of posts we want (5), plus one for our current post.
+				'date_query'             => [
+					[
+						'hour'   => 9,
+						'compare'=> '>=',
+					],
+					[
+						'hour'   => 17,
+						'compare'=> '<=',
+					]
```

Comment:
I thought for a while about this one. Date queries can be tricky. Unlike some other hacks (`post__not_in`), I'm not aware of a great way to make this more effective. If we could assume a small number of posts, we could fetch everything and then filter by date in PHP, but that felt like overkill without more context.

There's also the consideration of timezones and how that affects this query, but that's another can of worms.